src/main/resources/com/google/api/codegen/java/message.snip

```diff
@@ -6,18 +6,37 @@
 
   @@Generated("by GAPIC")
   @@BetaApi
-  public final class {@apiFile.schema.typeName} implements Serializable {
-    {@members(apiFile.schema)}
-    {@constructor(apiFile.schema)}
-    {@getters(apiFile.schema)}
-    {@newBuilder(apiFile.schema)}
-    {@builder(apiFile.schema)}
-    {@toString(apiFile.schema)}
-    {@equalsMethod(apiFile.schema)}
-    {@hashCode(apiFile.schema)}
+  {@schemaClassDef(apiFile.schema)}
+@end
+
+@private schemaClassDef(schema)
+  public final class {@schema.innerTypeName} implements Serializable {
+    {@members(schema)}
+
+    {@constructor(schema)}
+
+    {@getters(schema)}
+    {@innerSchemas(schema)}
+    {@newBuilder(schema)}
+
+    {@builder(schema)}
+
+    {@toString(schema)}
+
+    {@equalsMethod(schema)}
+
+    {@hashCode(schema)}
   }
 @end
 
+@private innerSchemas(schema)
+  @join property : schema.properties
+    @if {@property.properties}
+      {@schemaClassDef(property)}
```

Comment:
Right now, the compute discovery doc's most nested class is 3-deep (Operation.Warnings.Data)